src/python/pants/bin/goal_runner.py

```diff
@@ -153,73 +170,75 @@ def _setup_context(self):
       context = Context(options=self._options,
                         run_tracker=self._run_tracker,
                         target_roots=target_root_instances,
+                        v2_target_roots=target_roots,
```

Comment:
The fact that `Context` enters into the picture at all here is surprising to me. It seems like the context could/should just always be created _after_ all v2 logic has already completed?

Reply:
fixed.